odbc/src/api/data.rs

```diff
@@ -681,13 +681,13 @@ pub unsafe fn format_cached_data(
     mongo_handle: &mut MongoHandle,
     cached_data: CachedData,
     col_or_param_num: USmallInt,
-    target_type: CDataType,
+    target_type: SmallInt,
```

Comment:
To make it more clear, I think @nbagnard's idea is to create a helper function (where one doesn't already exist) with the same api as the current function and the SQLXXXX function will only convert the types to the proper enum and call the helper function.

Reply:
I was not even talking about having a helper (macro or otherwise) yet, but yes, this is ultimately what I'd like to go for.
The C/Rust interface, in our case the ODBC API functions, should be the only place where we have to handle this.
Everywhere else we can use enum values because we know they are not a "danger" anymore